spec/money_spec.rb

```diff
@@ -38,6 +38,14 @@
     expect(money.to_money).to eq(money)
   end
 
+  it "#to_money uses the provided currency when it doesn't already have one" do
+    expect(Money.new(1).to_money('CAD')).to eq(Money.new(1, 'CAD'))
+  end
+
+  it "#to_money doesn't overwrite the currency" do
+    expect(Money.new(1, 'USD').to_money('CAD')).to eq(Money.new(1, 'USD'))
```

Comment:
This might become a source of hard to debug bugs. Should we mark this behaviour deprecated?

Reply:
I also wonder if we want to warn (kinda like we do with `MoneyBag` anytime that currency param differs from the underlying currency?)